tests/client_test.py

```diff
@@ -1,148 +1,176 @@
 from datetime import datetime, timezone
-from unittest.mock import MagicMock, patch
+from unittest.mock import MagicMock, Mock, patch
 
 import pytest
 
 from riverqueue import Client, InsertOpts, UniqueOpts
+from riverqueue.driver import DriverProtocol, ExecutorProtocol
+import sqlalchemy
 
 from tests.simple_args import SimpleArgs
 
 
 @pytest.fixture
-def mock_driver():
-    return MagicMock()
+def mock_driver() -> DriverProtocol:
+    return MagicMock(spec=DriverProtocol)
 
 
 @pytest.fixture
-def client(mock_driver):
+def mock_exec(mock_driver) -> ExecutorProtocol:
+    def mock_context_manager(val) -> Mock:
+        context_manager_mock = MagicMock()
+        context_manager_mock.__enter__.return_value = val
+        context_manager_mock.__exit__.return_value = Mock()
+        return context_manager_mock
+
+    # def mock_context_manager(val) -> Mock:
+    #     return Mock(__enter__=val, __exit__=Mock())
```

Comment:
BTW, can you spot what's wrong with this version of the `mock_context_manager` compared to the one above? It looks like it should be right based on examples I've found, but it doesn't work. If it's not obvious, no worries, I'll just keep the uglier variant.